configure.ac

```diff
@@ -247,6 +268,27 @@ if test "x$enable_debug" = xyes; then
     fi
 fi
 
+# Checking for -fsanitize flags works differently from other compiler flags.
+# That's because with GCC there's an implicit linking dependency on libasan,
+# libtsan, etc. when using the related -fsanitize flags. Clang works
+# differently: it statically links these dependencies. Therefore we can't use
+# AC_CHECK_LIB to test this, we need to compile and link.
+if test "x$enable_asan" = xyes; then
```

Comment:
Does AX_CHECK_LINK_FLAG not do what we want here?

Reply:
That adds the flags to LDFLAGS, so no.

Reply:
Actually I think there's a trick to combine these (and remove the m4 file I added).